src/sql/src/plan/statement/show.rs

```diff
@@ -890,10 +890,20 @@ impl<'a> ShowColumnsSelect<'a> {
     }
 }
 
-fn simplify_names(catalog: &dyn SessionCatalog, sql: &str) -> Result<String, PlanError> {
+/// Convert a SQL statement into a form suitable for human consumption.
+fn humanize_sql(catalog: &dyn SessionCatalog, sql: &str) -> Result<String, PlanError> {
     let parsed = parse::parse(sql)?.into_element().ast;
     let (mut resolved, _) = names::resolve(catalog, parsed)?;
+
+    // Simplify names.
     let mut simplifier = NameSimplifier { catalog };
     simplifier.visit_statement_mut(&mut resolved);
+
+    // Strip internal `AS OF` syntax.
+    match &mut resolved {
+        Statement::CreateMaterializedView(stmt) => stmt.as_of = None,
+        _ => (),
+    }
```

Comment:
This change has two motivations:

* I think we (loosely?) follow the philosophy that `SHOW` outputs should be valid SQL you can execute.
* If we leave the `AS OF` in, all our tests that check `SHOW CREATE MATERIALIZED VIEW` output need to start worrying about a variable timestamp in the output.

Reply:
This SGTM.